vast/CMakeLists.txt

```diff
@@ -18,19 +14,26 @@ install(
 
 macro (create_tenzir_symlink name)
   add_custom_command(
-    OUTPUT "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/${name}"
+    TARGET tenzir POST_BUILD
```

Comment:
If you change from command output to a post-build target then you must list the created byproducts, otherwise the `clean` target cannot pick them up.

Reply:
I'll check if the handling of generator expressions in BYPRODUCT is correct, if it isn't I will leave it out.

NB: What is the use of the `clean` target when you can just delete the build tree? With the latter approach you will at least be sure that you won't get any problems from lingering cache variables.

Reply:
There are some situations in which you want to `cmake --build path/to/build --clean-first` because you changed your build configuration, e.g., when removing a plugin from the list of installed files. It's generally good practice to list all created files because of this.

I think the old approach for the symlink setup is much preferred because it only runs when needed, instead of running as a side-effect of another target completing. It should also likely be `POST_LINK` rather than `POST_BUILD`, otherwise the created symlink may not be usable within CMake if used as a dependency.

Reply:
I actually much prefer the attach to target approach. The old way you always had to build with `--target tenzir tenzir-ctl tenzir-node` to get everything that is needed. Easy to forget and get confusing errors.

Reply:
> I'll check if the handling of generator expressions in BYPRODUCT is correct, if it isn't I will leave it out.

To no surprise, it isn't:
```
CMake Error at vast/CMakeLists.txt:16 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_FILE_DIR:tenzir>

  No target "tenzir"
```

Reply:
Also, there is no `POST_LINK`, `POST_BUILD` means all other build steps have completed.

Reply:
I still don't understand the need to change this. Using the `TARGET_FILE_DIR` generator expression makes sense for maintainability, but if the goal is to just run `cmake --build path/to/build --target tenzir` to also run the `tenzir-ctl`, `tenzir-node`, and `vast` targets, then why not create an additional target for your convenience that depends on the all four? Something like this:

```cmake
add_custom_target(runtime DEPENDS tenzir tenzir-ctl tenzir-node vast)
```

Reply:
Because TARGET* generator expressions don't work in the OUTPUT field.
I found a way to make it work with the `clean` target, see the latest version.